src/modules/users/users.service.ts

```diff
@@ -68,8 +69,43 @@ export class UsersService {
     }
   }
 
-  async findAll(): Promise<User[]> {
-    return await this.usersRepository.find();
+  async findAll(filter?: SearchUserFilter | undefined): Promise<User[]> {
+    const contains = [];
+    if (Object.keys(filter).length) {
+      !filter.roles ? (filter.roles = Object.values(Roles)) : null;
+      !filter.roles.length ? filter.roles.push(...Object.values(Roles)) : null;
+
+      for (const role of filter.roles) {
+        const input = {
+          firstName: ILike(`%${filter.firstName}%`),
+          lastName: ILike(`%${filter.lastName}%`),
+          mobilePhone: ILike(`%${filter.mobilePhone}%`),
+          email: ILike(`%${filter.email}%`),
+          role: Roles[`${role}`],
+          createdAt: filter.createdAt,
+        };
+
+        Object.keys(input).map((key) => {
+          if (
+            !input[key] ||
+            input[key]._value === '%%' ||
+            input[key]._value === '%undefined%'
+          ) {
+            delete input[key];
+          }
+        });
+        contains.push({ ...input });
+      }
+    } else {
+      contains.push('');
+    }
```

Comment:
I will suggest explicitly adding fields from input to the result filter object (contains in the current context).
E.X.:
if(filter.firstName) {
   contains.firstName = ILike(`%${filter.firstName}%`);
}

As for roles, you can just omit the filtration if the argument was not passed.